src/cfclient/ui/tabs/FlightTab.py

```diff
@@ -355,19 +355,36 @@ def _change_input_labels(self, using_hover_assist):
         self.inputRollLabel.setText(roll)
         self.inputYawLabel.setText(yaw)
 
-    def _update_arm_button(self, connected):
+    def _update_supervisor_and_arming(self, connected):
         if not connected:
             self.armButton.setStyleSheet("")
             self.armButton.setText("Arm")
             self.armButton.setEnabled(False)
+            self._supervisor_state.setText("")
+            self._supervisor_state.setStyleSheet("")
+            self._supervisor_state.setText("")
             return
 
+        # We could set the text to "ready" or similar here, but an empty string is better for backwards compatibility as
+        # it would show "Ready" when locked for an older firmware.
+        self._supervisor_state.setText("")
+
         if self._is_flying():
             self.armButton.setEnabled(True)
             self.armButton.setText("Emergency stop")
             self.armButton.setStyleSheet("background-color: red")
+            self._supervisor_state.setText("Flying")
             return
 
+        if self._is_tumbled():
+            self._supervisor_state.setText("Tumbled")
+
+        if self._is_locked():
+            self._supervisor_state.setText("Locked - reboot")
```

Comment:
I tried this and the wording was a bit confusing. as it sseemed like it was locked and then was going to reboot by itself. It doesn't incite an action.

Perhaps 'Locked - please reboot' or someting?

Reply:
Good point! 
The space is limited though, the best I can come up with that fits is "Locked-pls reboot"

Reply:
I like the please part but perhaps something like "Locked -> reboot" looks better? That is not so clear that an action is needed though.